scripts/staging/fm-binclass.dml

```diff
@@ -0,0 +1,181 @@
+          # Compute training loss & accuracy
+          #loss_data   = cross_entropy_loss::forward(loss, y_batch);
+          loss_data = log_loss::forward(sfy, y_batch)
+          loss_reg_X  = l2_reg::forward(X, lambda); 
+          loss_reg_w0 = l2_reg::forward(w0, lambda);
+          loss_reg_W  = l2_reg::forward(W, lambda);
+          loss_reg_V  = l2_reg::forward(V, lambda);
+          
+          accuracy = (rowIndexMax(loss) == rowIndexMax(y_batch));
+          loss = loss_data + loss_reg_X + loss_reg_w0 + loss_reg_W + loss_reg_V;
+          
+          # Compute validation loss & accuracy
+          probs_val = predict(X_val)
+          #loss_val = cross_entropy_loss::forward(probs_val, y_val)
+          loss_val = log_loss::forward(probs_val, y_val)
+          accuracy_val = mean((probs<0.5) == (y_val<0.5))
+
+          # Output results
+          print("Epoch: " + e + ", Iter: " + i + ", Train Loss: " + loss + ", Train Accuracy: "
+                + accuracy + ", Val Loss: " + loss_val + ", Val Accuracy: " + accuracy_val)
+        }
+        
+        # 6.Send the result of sigmoid::forward and the correct labels y to log_loss::backward
+        dsfy = log_loss::backward(sfy, y);
+    
+        # 7.Send the above result through sigmoid::backward
+        dy = sigmoid::backward(dsfy, y);
+    
+        # 8.Send the above result through fm::backward
+        [dw0, dW, dV] = fm::backward(dy, X_batch, w0, W, V);
+        
+        # 9.Call adam::update for all parameters
+                
+        #X = adam::update(X, dX, lr, beta1, beta2, epsilon, t, mX, vX);
+        w0 = adam::update(w0, dw0, lr, beta1, beta2, epsilon, t, mw0, vw0);
+        W  = adam::update(W, dW, lr, beta1, beta2, epsilon, t, mW, vW );
+        V  = adam::update(V, dV, lr, beta1, beta2, epsilon, t, mV, vV );
+      }
+    }
+}
+
+
+predict = function(matrix[double] X, matrix[double] w0, matrix[double] W, matrix[double] V)
+    return (matrix[double] out) {
+  /*
+   * Computes the predictions for the given inputs.
+   *
+   * Inputs:
+   *  - X : n examples with d features, of shape (n, d).
+   *  - w0, W, V : trained model parameters. 
+   *  
+   * Outputs:
+   *  - out : target vector, y.
+   */
+    
+    # 1.initialize fm core
+    #[w0, W, V] = fm::init(d, k);
+    
+    # 2.Send inputs through fm::forward
+    y = fm::forward(X, w0, W, V);
+    
+    # 3.Send the above result through sigmoid::forward
+    out = sigmoid::forward(y);
+    
+    # 4.Send the above result through log_loss::forward
+    # loss = log_loss::forward(out); 
+  
+}
+
+eval = function(matrix[double] probs, matrix[double] y)
+    return (double loss, double accuracy) {
+   /*
+    * Computes loss and accuracy.
+    */
+    
+    # 1. just compute loss with log_loss::forward(probs, y)
+    loss = log_loss::forward(probs, y);
+
+    # 2. compute accuracy
+    #accuracy = mean((probs<0.5) == (y<0.5))
+    sqr_mean = mean( ( (probs<0.5) == (y<0.5) )^2 )
+    accuracy = sqr_mean^0.5 # rmse
```

Comment:
RMSE is great for a regression problem, but for this binary classification problem, it would be better to compute the accuracy as noted in a previous comment.